_posts/blog/2016-04-15-subscribe_on.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: "RxJava: subscribeOn vs observeOn"
+modified:
+categories: blog
+author: tomek_polanski
+excerpt:
+tags: [RxJava]
+image:
+date: 2016-04-14T13:00:55-01:00
+---
+
+
+## RxJava: subscribeOn vs observeOn
+
+One of the strongest aspects of RxJava is the simple way to schedule work on a desired thread using either ``subscribeOn`` or ``observeOn``. Understanding how each one works is crucial.  
+
+
+### observeOn:
+This method simply changes the thread of all operators further **downstream** (in the calls that come after). Let’s assume code is run from a UI thread: 
+
+<picture>
+	<img src="/images/ObserveOn.gif" alt="image">
+</picture>
+
+One of the most frequent misconceptions is that ``observeOn`` also acts upstream, but really it acts **only** downstream - things that happen after the ``observeOn`` call - unlike ``subscribeOn``.  
+
+### subscribeOn:
+This only influences the thread that is used when the observable is subscribed to and it will stay on it until changed.  
+One important fact is that ``subscribeOn`` does not work with ``Subject``s.  (We will return to this in a future post).  
+
+
+``` java
+just("Some String") // Computation
+.map(str -> str.length()) // Computation
+.map(length -> 2 * length) // Computation
+.subscribeOn(Schedulers.computation()) // -- changing the thread
+.subscribe(number -> Log.d("", "Number " + number)); // Computation
+```
+<br />
+**Position does not matter**
+``subscribeOn`` can be put in any place in the stream because it affects only the time of subscription. For example, the code from above is equal to this one:
+
+``` java
+just("Some String") // Computation
+    .subscribeOn(Schedulers.computation()) // the difference
+    .map(str -> str.length()) // Computation
+    .map(length -> 2 * length) // Computation
+    .subscribe(number -> Log.d("", "Number " + number)); // Computation
+```
+<br />
+**Methods that work with subscribeOn** 
```

Comment:
Should this be in the next blog? It seems to fit better with the discussion of which "Observables" obey the subscribeOn contract. If you keep this then move that mention of the Subjects not obeying the subscribeOn contract into here.

I think the word "work" is a little ambiguous too as I think of "doing work", instead use something like "operate in accordance with", "obeying the contact", "conforming" instead.